docs/src/main/asciidoc/writing-native-applications-tips.adoc

```diff
@@ -7,47 +7,85 @@ https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
 
 include::./attributes.adoc[]
 
-This guide contains various tips and tricks for getting around problems that might arise when attempting to run java applications as native executables.
+This guide contains various tips and tricks for getting around problems that might arise when attempting to run Java applications as native executables.
 
-== Register reflection
+Note that we differentiate two contexts where the solution applied might be different:
 
-Due to the fact that when building a native executable GraalVM operates with a closed world assumption (which enables it to remove all unused code, which in turns yields multiple benefits), reflection targets need to be explicitly declared.
+ * in the context of an application, you will rely on configuring the `native-image` configuration by tweaking your `pom.xml`;
+ * in the context of an extension, Quarkus offers a lot of infrastructure to simplify all of this.
 
-An example of failing to explicitly specify reflection is the following:
+Please refer to the appropriate section depending on your context.
 
-[source]
-----
-Caused by: org.xml.sax.SAXException: SAX2 driver class com.sun.org.apache.xerces.internal.parsers.SAXParser not found
-java.lang.ClassNotFoundException: com.sun.org.apache.xerces.internal.parsers.SAXParser
-at org.xml.sax.helpers.XMLReaderFactory.loadClass(XMLReaderFactory.java:230)
-----
+== Supporting native in your application
 
-If one were to execute the native executable generation process manually, the `-H:ReflectionConfigurationFiles=` flag would be used to point to a JSON configuration file that specifies the program elements that would be accessed reflectively.
+GraalVM comes with a number of constraints and making your application a native executable might require a few tweaks.
```

Comment:
`GraalVM comes with a number` -> `GraalVM imposes a number` maybe?

`and making` -> `so making` perhaps?

Reply:
applied the first one but not the second one. Don't know why but `and` sounds better to my ears.

Reply:
OK :)